src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -1008,6 +1008,185 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
+                using (writer.Operation(methodCallExpression))
+                {
+                    int argNum = methodCallExpression.Arguments.Count;
+                    writer.Write("includetimeseries(");
+                    for (var i = 0; i < argNum; i++)
+                    {
+                        context.Visitor.Visit(methodCallExpression.Arguments[i]);
+                        if (i < argNum - 1)
+                            writer.Write(",");
+                    }
+                    writer.Write(")");
+                }
+            }
+        }
+
+        public class IncludeCounterSupport : JavascriptConversionExtension
+        {
+            public static readonly IncludeCounterSupport Instance = new IncludeCounterSupport();
+
+            public override void ConvertToJavascript(JavascriptConversionContext context)
+            {
+                var methodCallExpression = context.Node as MethodCallExpression;
+
+                if (methodCallExpression?.Method.Name != nameof(RavenQuery.IncludeCounters))
+                    return;
+
+                if (methodCallExpression.Method.DeclaringType != typeof(RavenQuery))
+                    return;
+                if (methodCallExpression.Arguments.Count != 2)
+                    return;
+
+                context.PreventDefault();
+
+                var writer = context.GetWriter();
+                using (writer.Operation(methodCallExpression))
+                {
+                    int argNum = methodCallExpression.Arguments.Count;
+                    writer.Write("includecounters(");
+                    for (var i = 0; i < argNum; i++)
+                    {
+                        context.Visitor.Visit(methodCallExpression.Arguments[i]);
+                        if (i < argNum - 1)
+                            writer.Write(",");
+                    }
+                    writer.Write(")");
+                }
+            }
+        }
+
+        public class IncludeAllCounterSupport : JavascriptConversionExtension
+        {
+            public static readonly IncludeAllCounterSupport Instance = new IncludeAllCounterSupport();
+
+            public override void ConvertToJavascript(JavascriptConversionContext context)
+            {
+                var methodCallExpression = context.Node as MethodCallExpression;
+
+                if (methodCallExpression?.Method.Name != nameof(RavenQuery.IncludeAllCounters))
+                    return;
+
+                if (methodCallExpression.Method.DeclaringType != typeof(RavenQuery))
+                    return;
+                if (methodCallExpression.Arguments.Count != 1)
+                    return;
+
+                context.PreventDefault();
+
+                var writer = context.GetWriter();
+                using (writer.Operation(methodCallExpression))
+                {
+                    int argNum = methodCallExpression.Arguments.Count;
+                    writer.Write("includeallcounters(");
```

Comment:
`includes.counters(document)`

Reply:
Deleted IncludeAllCounterSupport class because we made the same call in includeCountersSupport and it'll already be made from there.
done